Client/Ecosia/UI/EcosiaFindInPageBar.swift

```diff
@@ -0,0 +1,238 @@
+        
+        addSubview(searchView)
+        searchView.addSubview(searchTextField)
+        searchView.addSubview(matchCountLabel)
+        addSubview(previousButton)
+        addSubview(nextButton)
+        addSubview(closeButton)
+        addSubview(topBorder)
+
+        applyTheme()
+        setupConstraints()
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(applyTheme), name: .DisplayThemeChanged, object: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    @discardableResult override func becomeFirstResponder() -> Bool {
+        searchTextField.becomeFirstResponder()
+        return super.becomeFirstResponder()
+    }
+    
+    @objc private func applyTheme() {
+        backgroundColor = .theme.ecosia.secondaryBackground
+        searchView.backgroundColor = .theme.ecosia.tertiaryBackground
+        searchTextField.textColor = .theme.ecosia.primaryText
+        matchCountLabel.textColor = .theme.ecosia.secondaryText
+        previousButton.tintColor = .theme.ecosia.primaryIcon
+        nextButton.tintColor = .theme.ecosia.primaryIcon
+        closeButton.setTitleColor(.theme.ecosia.primaryButton, for: .normal)
+        topBorder.backgroundColor = .theme.ecosia.border
+    }
+    
+    private func setupConstraints() {
+        self.snp.makeConstraints { make in
+            make.height.equalTo(EcosiaFindInPageBar.barHeight)
+        }
+        
+        searchView.snp.makeConstraints { make in
+            make.leading.equalToSuperview().offset(16)
+            make.top.equalToSuperview().offset(EcosiaFindInPageBar.searchViewTopBottomSpacing)
+            make.bottom.equalToSuperview().inset(EcosiaFindInPageBar.searchViewTopBottomSpacing)
+        }
+        
+        searchTextField.snp.makeConstraints { make in
+            make.leading.equalToSuperview().offset(16)
+            make.centerY.equalToSuperview()
+        }
+        searchTextField.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        searchTextField.setContentCompressionResistancePriority(.defaultLow, for: .horizontal)
+
+        matchCountLabel.snp.makeConstraints { make in
+            make.leading.equalTo(searchTextField.snp.trailing)
+            make.trailing.equalToSuperview().inset(13)
+            make.centerY.equalToSuperview()
+        }
+        matchCountLabel.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        matchCountLabel.setContentCompressionResistancePriority(.defaultHigh, for: .horizontal)
+
+        previousButton.snp.makeConstraints { make in
+            make.leading.equalTo(searchView.snp.trailing).offset(14)
+            make.centerY.equalToSuperview()
+        }
+
+        nextButton.snp.makeConstraints { make in
+            make.leading.equalTo(previousButton.snp.trailing).offset(29)
```

Comment:
same for the hardcoded values being put in a centralized place (that `UX` struct perhaps)